src/kv/KeyValueDB.h

```diff
@@ -205,6 +206,7 @@ class KeyValueDB {
       return "";
     }
     virtual ceph::buffer::list value() = 0;
+    virtual std::string_view value_as_sv() = 0;
```

Comment:
Would be nice to kill old variants ultimately. Please note that doing the `memcpy` here results in more fragmentation of `bufferlist` at the `Message` layer.

Reply:
Keys should undergo exactly the same expansion procedure.

There is a problem with `string_view` lifetime. It should be clearly stated for how long the value existence is guaranteed. 
For RocksDB that would be until next `seek/bound/next` call, I am reasonably sure.
For MemStore and KStore it should be checked.